### DISCUSSION TITLE
Uninterruptible power supply

### DISCUSSION THREAD

**Discussion by spestana** [Ribbit-Network/ribbit-network-frog-hardware#78] [Ideas]
This may be related to discussions over in #52 .

I had some brief power outages in recent storms, which reset the frog sensor requiring me to connect it to wifi again (I hadn't noticed that it had been knocked offline at first!) This made me think that adding a small battery in the loop could prevent this from happening in the future. 

A quick search online turned up lots of mini uninterruptible power supplies which could be easy enough to implement external to the sensor, but I also have [one of these power supply circuits + battery](https://www.adafruit.com/product/1944) on hand I could try out, and then there's the possibility of something more integrated with the sensor as a whole [maybe along these lines](https://www.beyondlogic.org/ltc4040-5v-2-5a-uninterruptible-power-supply-ups-with-usb-c/).

I'm opening this discussion to see if others have ideas, and/or to document whatever I end up trying out.

**Comment by keenanjohnson** [comment]
I think this is great. It feels like it could be a nice incremental step towards a fully solar + battery Frog.

I know @eren-rudy had started doing some thinking about the power system looking forward to a solar-powered sensor. Perhaps he might have some thoughts here.

**Comment by eren-rudy** [comment]
Great idea @spestana.

Maybe this brings up a relevant discussion regarding batteries, which would apply to both the UPS for the wall-powered version, and the solar-powered version. Considering the shipping limitations of LiPo batteries, I think it makes a lot of sense to use a standard battery pack such as AA, so its great to see this implemented in that [USB-C UPS](https://www.beyondlogic.org/ltc4040-5v-2-5a-uninterruptible-power-supply-ups-with-usb-c/) that @spestana linked. That will probably be a great reference point for the solar-powered sensor.

As an alternative to requiring a UPS, would it be possible to have the sensor automatically connect to WiFi after re-gaining power? I'm guessing there wouldn't be a way within Balena to somehow store the user-entered WiFi credentials in EEPROM, so perhaps we could implement our own sensor setup workflow using Bluetooth (and maybe this could be related to the local-debug interface project) where the user enters the required credentials on their phone and we can somehow store them securely for Balena to use upon re-boot. I'm not really an expert on this stuff, so maybe I'm way off base in terms of the feasibility of a plan like this. Would be curious to hear thoughts on this. @keenanjohnson @djgood.

**Comment by keenanjohnson** [comment]
Do you mean reschargeable AA's @eren-rudy?

@eren-rudy Balena already stores wifi credentials and will reconnect after powering back up.

**Reply by eren-rudy** [reply]
@keenanjohnson Yes, I meant rechargeable AAs. We definitely would have to be aware of the possibility of people plugging in alternate chemistries, but other than that I think it might be a good solution.

Ah! I didn't know that. Do you know what the issue that @spestana is describing is, then? It seems I don't have a great understanding of the issues that arise when the device loses power (other than temporarily not being able to collect data).
